src/reader.rs

```diff
@@ -511,6 +481,12 @@ pub struct Sample {
     pub duration: u64,
 }
 
+impl Sample {
+    pub fn range(&self) -> std::ops::Range<usize> {
```

Comment:
should be documented that this is a range into the bytes of the buffer. Without the context of this PR I would have had no clue. Also, needs some more fitting name `byte_range_in_input_data` or similar?

Reply:
The whole library is missing many doc comments, so I'm not going to go through absolutely everything, but added some docs to this method and some related fields as a drive-by.

Reply:
> The whole library is missing many doc comments

https://en.wikipedia.org/wiki/Broken_windows_theory